packages/babel-plugin-transform-typescript/src/index.ts

```diff
@@ -307,24 +328,33 @@ export default declare((api, opts) => {
           )
         ) {
           path.remove();
+          return;
         }
+
+        IMPLICITLY_ESM.add(state.file.ast.program);
       },
 
-      ExportSpecifier(path) {
+      ExportSpecifier(path, state) {
         // remove type exports
         if (!path.parent.source && isGlobalType(path, path.node.local.name)) {
           path.remove();
+          return;
         }
+
+        IMPLICITLY_ESM.add(state.file.ast.program);
```

Comment:
Q: Why do we need this for `ExportSpecifier`?

Reply:
Yeah good point — looks like this is already handled above.